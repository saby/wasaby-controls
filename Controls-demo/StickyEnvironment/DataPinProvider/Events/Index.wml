<div class="controls-demo_Wrapper controlsDemo_fixedWidth500">
   <div data-qa="controls-demo__text_top">Top edge data id: {{_topEdgeData}}</div>
   <div data-qa="controls-demo__text_bottom">Bottom edge data id: {{_bottomEdgeData}}</div>

   <Controls.stickyEnvironment:DataPinProvider on:edgesDataChanged="_onEdgesDataChanged()">
      <Controls.scroll:Container attr:style="height: 500px;">
         <Controls.list:View
            source="{{_viewSource}}"
            keyProperty="key"
            navigation="{{_navigation}}"
         >
            <ws:virtualScrollConfig page="{{50}}" pageSize="{{50}}"/>
            <ws:itemTemplate>
               <ws:partial template="Controls/list:ItemTemplate">
                  <ws:contentTemplate>
                     <Controls.stickyEnvironment:DataPinContainer data="{{itemTemplate.item.contents.get('key')}}">
                        <strong>Запись с id="{{itemTemplate.item.contents.get('key')}}"</strong>
                     </Controls.stickyEnvironment:DataPinContainer>

                     {{itemTemplate.item.contents.get('title')}}
                     <i data-qa="Controls-Demo_StickyEnvironment__hooks">h</i>
                  </ws:contentTemplate>
               </ws:partial>
            </ws:itemTemplate>
         </Controls.list:View>
      </Controls.scroll:Container>
   </Controls.stickyEnvironment:DataPinProvider>
</div>
