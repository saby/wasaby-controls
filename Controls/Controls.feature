<?xml version="1.0" encoding="UTF-8" ?>
<features version="1.00">

  <feature id="filter_search_new" project="https://project.sbis.ru/uuid/b210aea0-dc6e-4dc1-bf0f-7a5adadf2daf/page/project-main" responsible="Золотова Э.Е." responsible_uuid="55dd1897-4ba4-4c0e-b17e-e05fc0a86623">
    <description>Включает фильтр с поиском в реестрах</description>
  </feature>

  <feature id="menu_mode_selector" project="https://online.sbis.ru/opendoc.html?guid=22249abd-c279-4400-a71a-a1b986bf2a0c&amp;client=3" responsible="Золотова Э.Е." responsible_uuid="55dd1897-4ba4-4c0e-b17e-e05fc0a86623">
    <description>Включает отображение меню в виде справочника</description>
  </feature>

</features>
