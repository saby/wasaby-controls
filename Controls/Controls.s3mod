<?xml version="1.0" encoding="UTF-8" ?>
<ui_module id="f6fdf353-3ab3-4dbb-8671-01c2b418bf0a" kaizen_zone="916d4071-6bb4-4800-b3ff-2ee6725c9fdc"  name="Controls" package="Wasaby" responsible="Крайнов Д.О." responsible_uuid="0a464b92-3129-49f2-8a5e-7ce6ecfa37b0" required="1" version="1.00">

  <description>Готовый набор визуальных компонентов, отвечающих корпоративным стандартам. Позволяют быстро спроектировать визуальную составляющую веб-приложения.</description>

    <depends>
      <module id="686f183d-0bcc-4732-b4e4-6016c2a67a0f" name="Types"/>
      <module id="323fd54f-1fab-42a6-8aea-6726d6392d53" name="Application"/>
      <module id="63e1a7e5-4d75-42fa-91a3-977dc90ff1e1" name="SbisEnv"/>
      <module id="0df0fce3-53ba-47f8-987d-edf1fd078295" name="WS.Core"/>
      <module id="636177e5-4d75-44fa-9173-977dc90ff111" name="Env"/>
      <module id="4220f331-8217-437b-810c-8c4479323a27" name="EnvConfig"/>
      <module id="2274d173-fe30-4fcf-81ca-767915919d69" name="EnvTouch"/>
      <module id="656177e2-4d75-42fa-9173-977dc90ff121" name="Browser"/>
      <module id="10efeac7-0f02-4c73-9474-0d1d0db077b6" name="View"/>
      <module id="0f5d26db-dd61-4957-9375-8861ad66f219" name="Vdom"/>
      <module id="f27abe12-19df-4c9f-b886-634bd20a0011" name="UI"/>
      <module id="832cda0a-e19c-4ffc-a1d1-47edbb6b7501" name="I18n"/>
      <module id="37e402f1-2fb0-4b73-81c5-113cf94dbf1c" name="WasabyLoader"/>
      <module id="e4bbd0f2-f7ce-404e-88b2-5130d105b770" name="ParametersWebAPI"/>
      <module id="6ae4f1d6-9ecf-4805-a98c-65e47906eedd" name="Tailwind"/>
      <module id="50f80520-0983-4049-b17a-d893bce91d69" name="ErrorHandling"/>
      <module id="8298f3a2-1964-11ed-861d-0242ac120002" name="Controls-DataEnv"/>
    </depends>

  <load_after/>

  <parameters/>

  <resources/>

  <preload/>

</ui_module>
