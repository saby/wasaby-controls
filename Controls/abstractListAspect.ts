export {
    AbstractAspectStateManager,
    TAbstractAspectStateManagerProps,
} from 'Controls/ListAspects/_abstractListAspect/abstract/AbstractAspectStateManager';
export {
    IListChange,
    ListChangeNameEnum,
    IGetListChangeByName,
    IMoveMarkerListChange,
} from 'Controls/ListAspects/_abstractListAspect/common/ListChanges';

export {
    IStateWithItems,
    copyStateWithItems,
} from 'Controls/ListAspects/_abstractListAspect/common/IStateWithItems';
export {
    copyStateWithHierarchyItems,
    IStateWithHierarchyItems,
} from 'Controls/ListAspects/_abstractListAspect/common/IStateWithHierarchyItems';

export * as CommonUILogic from 'Controls/ListAspects/_abstractListAspect/common/UILogic';
