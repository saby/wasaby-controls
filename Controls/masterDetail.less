@import 'Controls/_masterDetail/Base/_Base.less';
