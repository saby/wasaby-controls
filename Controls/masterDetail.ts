/**
 * @kaizen_zone 98d4b42e-2c0e-4268-a9e8-1e54d6e8ef27
 */
/**
 * Библиотека контролов для организации двухколоночных списков, в которых выбор элемента из первой колонки влияет на содержимое второй колонки.
 * @library
 * @includes Base Controls/_masterDetail/Base
 * @includes List Controls/_masterDetail/List
 * @public
 */

/*
 * masterDetail library
 * @library
 * @includes Base Controls/_masterDetail/Base
 * @public
 * @author Герасимов А.М.
 */

export { default as Base } from 'Controls/_masterDetail/Base';
export { default as List } from 'Controls/_masterDetail/List';
export { TMasterVisibility, IMasterWidth } from 'Controls/_masterDetail/Base';
