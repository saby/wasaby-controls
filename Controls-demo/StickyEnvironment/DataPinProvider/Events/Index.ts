import { Memory } from 'Types/source';
import { SyntheticEvent } from 'UI/Vdom';
import { IEdgesData } from 'Controls/stickyEnvironment';
import { Control, IControlOptions, TemplateFunction } from 'UI/Base';
import { generateData } from 'Controls-demo/list_new/DemoHelpers/DataCatalog';
import * as template from 'wml!Controls-demo/StickyEnvironment/DataPinProvider/Events/Index';

export default class Index extends Control<IControlOptions> {
    protected _template: TemplateFunction = template;

    protected _topEdgeData: number;
    protected _bottomEdgeData: number;

    protected _viewSource: Memory;
    protected _navigation: object = {
        source: 'page',
        view: 'infinity',
        sourceConfig: {
            page: 10,
            pageSize: 50,
            hasMore: false,
        },
    };

    private _dataArray: { key: number; title: string }[] = generateData<{
        key: number;
        title: string;
    }>({
        count: 1000,
        entityTemplate: { title: 'lorem' },
    });

    protected _beforeMount(): void {
        this._viewSource = new Memory({
            keyProperty: 'key',
            data: this._dataArray,
        });
    }

    protected _onEdgesDataChanged(
        event: SyntheticEvent,
        data: IEdgesData<number>
    ): void {
        this._topEdgeData = data.top.above || data.top.below;
        this._bottomEdgeData = data.bottom.above;
    }
}
