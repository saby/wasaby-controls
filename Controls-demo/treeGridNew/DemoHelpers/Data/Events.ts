import { IColumn } from 'Controls/grid';
import { IData } from '../Interface';

export const Events = {
    getColumns: (): IColumn[] => {
        return [
            {
                displayProperty: 'title',
                width: '',
            },
        ];
    },
    getDataBeforeExpanded: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataBeforeCollapsed: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataCollapsedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataExpandedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
};
