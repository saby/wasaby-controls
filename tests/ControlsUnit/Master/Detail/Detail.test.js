define(['Controls/_masterDetail/Base'], function (masterDetail) {
   'use strict';
   describe('Controls.Container.MasterDetail', function () {
      it('selected master value changed', () => {
         let Control = new masterDetail.default();
         let event = {
            stopPropagation: jest.fn()
         };
         Control._selectedMasterValueChangedHandler(event, 'newValue');
         expect(Control._selected).toEqual('newValue');
         Control.destroy();
      });

      it('beforeMount', (done) => {
         const Control = new masterDetail.default();
         Control._getSettings = () => {
            return Promise.resolve({ 1: 500 });
         };

         let options = {
            propStorageId: '1',
            masterMinWidth: 100,
            masterWidth: 200,
            masterMaxWidth: 300
         };

         Control._beforeMount(options).then((result) => {
            expect(result).toEqual('300px');
            expect(Control._currentMinWidth).toEqual('100px');
            expect(Control._currentMaxWidth).toEqual('300px');
            Control.destroy();
            done();
         });
      });

      it('initCurrentWidth', () => {
         let Control = new masterDetail.default();
         let options = {
            propStorageId: '1',
            masterMinWidth: 0,
            masterWidth: 0,
            masterMaxWidth: 0
         };
         Control.initCurrentWidth(options.masterWidth);
         expect(Control._currentWidth).toEqual('0px');
         Control.destroy();
      });

      it('update offset', () => {
         let Control = new masterDetail.default();

         // base
         let options = {
            masterMinWidth: 100,
            masterWidth: 200,
            masterMaxWidth: 299,
            propStorageId: 10
         };

         Control._canResizing = Control._isCanResizing(options);
         Control._updateOffset(options);
         expect(Control._minOffset).toEqual(100);
         expect(Control._maxOffset).toEqual(99);
         expect(Control._currentWidth).toEqual('200px');

         Control._beforeUpdate(options);
         expect(Control._minOffset).toEqual(100);
         expect(Control._maxOffset).toEqual(99);
         expect(Control._currentWidth).toEqual('200px');

         // wrong maxWidth
         options = {
            masterMinWidth: 100,
            masterWidth: 200,
            masterMaxWidth: 150
         };
         Control._beforeUpdate(options);
         expect(Control._minOffset).toEqual(50);
         expect(Control._maxOffset).toEqual(0);
         expect(Control._currentWidth).toEqual('150px');

         // wrong minWidth
         options = {
            masterMinWidth: 250,
            masterWidth: 200,
            masterMaxWidth: 299
         };
         Control._beforeUpdate(options);
         expect(Control._minOffset).toEqual(0);
         expect(Control._maxOffset).toEqual(99);
         expect(Control._currentWidth).toEqual('250px');

         options = {
            masterMinWidth: 0,
            masterWidth: 0,
            masterMaxWidth: 0
         };
         Control._beforeUpdate(options);
         expect(Control._minOffset).toEqual(0);
         expect(Control._maxOffset).toEqual(0);
         expect(Control._currentMinWidth).toEqual('0px');
         expect(Control._currentMaxWidth).toEqual('0px');
         expect(Control._currentWidth).toEqual('0px');

         Control.destroy();
      });

      it('_beforeUpdate with new propStorageId', async () => {
         let Control = new masterDetail.default();
         let options = {
            ...masterDetail.default.getDefaultOptions(),
            propStorageId: 'test'
         };
         Control._beforeMount(options);
         options.propStorageId = 'newTest';
         Control._getSettings = () => {
            const result = {
               newTest: 50
            };
            return Promise.resolve(result);
         };
         await Control._beforeUpdate(options);
         expect(Control._currentWidth).toEqual('50px');
         Control.destroy();
      });

      it('_dragStartHandler', () => {
         let Control = new masterDetail.default();
         let options = {
            masterMinWidth: 100,
            masterWidth: 150,
            masterMaxWidth: 200
         };
         Control._options = options;
         Control._canResizing = true;
         Control._dragStartHandler();
         expect(Control._minOffset).toEqual(50);
         expect(Control._maxOffset).toEqual(50);
         expect(Control._currentWidth).toEqual('150px');
         Control.destroy();
      });

      it('is can resizing', () => {
         let Control = new masterDetail.default();
         let options = {
            masterMinWidth: 250,
            masterWidth: 200,
            masterMaxWidth: 299,
            propStorageId: 10
         };
         expect(Control._isCanResizing(options)).toEqual(true);

         options.masterMinWidth = options.masterMaxWidth;
         expect(Control._isCanResizing(options)).toEqual(false);
         Control.destroy();
      });

      it('afterRender', () => {
         const Control = new masterDetail.default();
         let isStartRegister = false;
         Control._startResizeRegister = () => {
            isStartRegister = true;
         };

         Control._afterRender();
         expect(isStartRegister).toEqual(false);

         Control._currentWidth = 1;
         Control._afterRender();
         expect(isStartRegister).toEqual(true);

         Control.destroy();
      });

      it('_afterUpdate', () => {
         const Control = new masterDetail.default();
         const oldOptions = {
            masterVisibility: false
         };
         let isStartRegister = false;
         Control._options = {
            masterVisibility: false
         };
         Control._startResizeRegister = () => {
            isStartRegister = true;
         };

         Control._afterUpdate(oldOptions);
         expect(isStartRegister).toBe(false);

         Control._options.masterVisibility = true;
         Control._afterUpdate(oldOptions);
         expect(isStartRegister).toBe(true);

         Control.destroy();
      });

      it('_resizeHandler with propStorageId', () => {
         const Control = new masterDetail.default();
         let isUpdateOffset = false;
         Control._startResizeRegister = jest.fn();
         Control._updateOffsetDebounced = () => {
            isUpdateOffset = true;
         };

         Control._options = {
            propStorageId: '123'
         };
         Control._container = {
            closest: () => {
               return false;
            }
         };
         Control._resizeHandler();
         expect(isUpdateOffset).toBe(true);

         Control.destroy();
      });

      it('_resizeHandler without propStorageId', () => {
         const Control = new masterDetail.default();
         let isUpdateOffset = false;
         Control._updateOffsetDebounced = () => {
            isUpdateOffset = true;
         };
         Control._startResizeRegister = jest.fn();

         Control._container = {
            closest: () => {
               return false;
            }
         };
         Control._resizeHandler();
         expect(isUpdateOffset).toBe(false);

         Control.destroy();
      });

      it('_startResizeRegister without resizeDetectMaster', () => {
         const Control = new masterDetail.default();
         Control._options = {
            masterVisibility: 'hidden'
         };
         let errorRaised = false;
         Control._children = {
            resizeDetectDetail: {
               start: jest.fn()
            }
         };
         try {
            Control._startResizeRegister();
         } catch (e) {
            errorRaised = true;
         }
         expect(errorRaised).toBe(false);
      });

      it('masterWidthChanged', () => {
         const control = new masterDetail.default();
         const event = {};
         const offset = 100;
         let isSetSettings = false;
         jest.spyOn(control, '_notify').mockClear().mockImplementation();
         control._currentWidth = 100;
         control._setSettings = () => {
            isSetSettings = true;
         };

         control._offsetHandler(event, offset);
         expect(isSetSettings).toEqual(true);

         expect(control._notify).toHaveBeenCalledWith('masterWidthChanged', ['200px']);

         jest.restoreAllMocks();
      });

      it('touch resize', () => {
         const control = new masterDetail.default();
         let isStartDragCalled = false;
         let isEndDragCalled = false;
         let moveOffset = 0;
         let endMoveOffset = 0;
         control._children = {
            resizingLine: {
               startDrag: () => {
                  isStartDragCalled = true;
               },
               drag: (offset) => {
                  moveOffset = offset;
               },
               endDrag: (offset) => {
                  isEndDragCalled = true;
                  endMoveOffset = offset;
               }
            }
         };
         const getFakeEvent = (pageX, target, currentTarget) => {
            return {
               target,
               currentTarget,
               nativeEvent: {
                  changedTouches: [
                     {
                        pageX
                     }
                  ]
               }
            };
         };
         control._needHandleTouch = () => {
            return false;
         };
         control._touchstartHandler(getFakeEvent(1, 2, 3));
         control._touchendHandler(getFakeEvent());
         expect(isStartDragCalled).toEqual(false);
         expect(isEndDragCalled).toEqual(false);

         control._needHandleTouch = () => {
            return true;
         };
         control._canResizing = true;
         control._touchstartHandler(getFakeEvent(10, 'body', 'body'));
         control._touchMoveHandler(getFakeEvent(20, 'body', 'body'));
         control._touchendHandler(getFakeEvent(31, 'body', 'body'));
         expect(moveOffset).toEqual(10);
         expect(endMoveOffset).toEqual(21);

         control.destroy();
      });
   });
});
