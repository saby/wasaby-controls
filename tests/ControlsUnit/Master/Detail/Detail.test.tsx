/**
 * @jest-environment jsdom
 */
import * as React from 'react';
import { Base as MasterDetail } from 'Controls/masterDetail';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { WasabyEvents } from 'UICore/Events';

const faceContentTemplate = React.forwardRef<{}, HTMLDivElement>((_, ref) => <div ref={ref}></div>);

type MasterDetailPropsPart = Partial<React.ComponentProps<typeof MasterDetail>>;

const masterDetailDefaultProps: React.ComponentProps<typeof MasterDetail> = {
    master: faceContentTemplate,
    detail: faceContentTemplate,
};

describe('Controls/masterDetail:Base', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        WasabyEvents.initInstance(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    it('MasterDetail рендер с установкой макс/мин ширины (base)', () => {
        // base
        const props: MasterDetailPropsPart = {
            masterMinWidth: 100,
            masterWidth: 200,
            masterMaxWidth: 299,
        };

        act(() => {
            render(<MasterDetail {...masterDetailDefaultProps} {...props} />, container);
        });

        expect(container).toMatchSnapshot();
    });

    it('MasterDetail рендер с установкой макс/мин ширины (wrong maxWidth)', () => {
        // wrong maxWidth
        const props: MasterDetailPropsPart = {
            masterMinWidth: 100,
            masterWidth: 200,
            masterMaxWidth: 150,
        };

        act(() => {
            render(<MasterDetail {...masterDetailDefaultProps} {...props} />, container);
        });

        expect(container).toMatchSnapshot();
    });

    it('MasterDetail рендер с установкой макс/мин ширины (wrong minWidth)', () => {
        // wrong minWidth
        const props: MasterDetailPropsPart = {
            masterMinWidth: 250,
            masterWidth: 200,
            masterMaxWidth: 299,
        };

        act(() => {
            render(<MasterDetail {...masterDetailDefaultProps} {...props} />, container);
        });

        expect(container).toMatchSnapshot();
    });

    it('MasterDetail рендер с установкой masterVisibility', () => {
        const props: MasterDetailPropsPart = {
            masterVisibility: 'hidden',
        };

        act(() => {
            render(<MasterDetail {...masterDetailDefaultProps} {...props} />, container);
        });

        expect(container).toMatchSnapshot();
    });

    it('MasterDetail рендер с установкой masterPosition', () => {
        const props1: MasterDetailPropsPart = {
            masterPosition: 'right',
        };
        const props2: MasterDetailPropsPart = {
            masterPosition: 'right',
        };

        act(() => {
            render(<MasterDetail {...masterDetailDefaultProps} {...props1} />, container);
        });
        expect(container).toMatchSnapshot();

        act(() => {
            render(<MasterDetail {...masterDetailDefaultProps} {...props2} />, container);
        });
        expect(container).toMatchSnapshot();
    });

    it('MasterDetail рендер с установкой ширины + propStorageId', () => {
        const props: MasterDetailPropsPart = {
            masterMinWidth: 100,
            masterWidth: 200,
            masterMaxWidth: 299,
            propStorageId: 'testId',
            initialMasterWidth: 200,
        };

        act(() => {
            render(<MasterDetail {...masterDetailDefaultProps} {...props} />, container);
        });

        expect(container).toMatchSnapshot();
    });
});
