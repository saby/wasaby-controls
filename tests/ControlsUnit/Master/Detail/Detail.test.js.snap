// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls/masterDetail:Base MasterDetail рендер с установкой masterPosition 1`] = `
<div>
  <div
    class="
                controls-MasterDetail controls-MasterDetail_masterPosition-right
                
                ws-flexbox controls_list_theme-undefined
                
                undefined
              "
  >
    <div
      class="
                        controls-MasterDetail_master undefined
                        controls-MasterDetail_master-contrast controls-MasterDetail_master_width
                      "
      style="width: 27%; max-width: 50%; min-width: 30px;"
    >
      <div />
    </div>
    <div
      class="
                            controls-MasterDetail_details
                            controls-MasterDetail_details-contrast
                            ws-flexbox ws-flex-grow-1 
                          "
    >
      <div />
    </div>
    <div
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Controls/masterDetail:Base MasterDetail рендер с установкой masterPosition 2`] = `
<div>
  <div
    class="
                controls-MasterDetail controls-MasterDetail_masterPosition-right
                
                ws-flexbox controls_list_theme-undefined
                
                undefined
              "
  >
    <div
      class="
                        controls-MasterDetail_master undefined
                        controls-MasterDetail_master-contrast controls-MasterDetail_master_width
                      "
      style="width: 27%; max-width: 50%; min-width: 30px;"
    >
      <div />
    </div>
    <div
      class="
                            controls-MasterDetail_details
                            controls-MasterDetail_details-contrast
                            ws-flexbox ws-flex-grow-1 
                          "
    >
      <div />
    </div>
    <div
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Controls/masterDetail:Base MasterDetail рендер с установкой masterVisibility 1`] = `
<div>
  <div
    class="
                controls-MasterDetail controls-MasterDetail_masterPosition-left
                
                ws-flexbox controls_list_theme-undefined
                
                undefined
              "
  >
    <div
      class="
                            controls-MasterDetail_details
                            controls-MasterDetail_details-contrast
                            ws-flexbox ws-flex-grow-1 
                          "
    >
      <div />
    </div>
    <div
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Controls/masterDetail:Base MasterDetail рендер с установкой макс/мин ширины (base) 1`] = `
<div>
  <div
    class="
                controls-MasterDetail controls-MasterDetail_masterPosition-left
                
                ws-flexbox controls_list_theme-undefined
                
                undefined
              "
  >
    <div
      class="
                        controls-MasterDetail_master undefined
                        controls-MasterDetail_master-contrast controls-MasterDetail_master_width
                      "
      style="width: 200px; max-width: 299px; min-width: 100px;"
    >
      <div />
    </div>
    <div
      class="
                            controls-MasterDetail_details
                            controls-MasterDetail_details-contrast
                            ws-flexbox ws-flex-grow-1 
                          "
    >
      <div />
    </div>
    <div
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Controls/masterDetail:Base MasterDetail рендер с установкой макс/мин ширины (wrong maxWidth) 1`] = `
<div>
  <div
    class="
                controls-MasterDetail controls-MasterDetail_masterPosition-left
                
                ws-flexbox controls_list_theme-undefined
                
                undefined
              "
  >
    <div
      class="
                        controls-MasterDetail_master undefined
                        controls-MasterDetail_master-contrast controls-MasterDetail_master_width
                      "
      style="width: 200px; max-width: 150px; min-width: 100px;"
    >
      <div />
    </div>
    <div
      class="
                            controls-MasterDetail_details
                            controls-MasterDetail_details-contrast
                            ws-flexbox ws-flex-grow-1 
                          "
    >
      <div />
    </div>
    <div
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Controls/masterDetail:Base MasterDetail рендер с установкой макс/мин ширины (wrong minWidth) 1`] = `
<div>
  <div
    class="
                controls-MasterDetail controls-MasterDetail_masterPosition-left
                
                ws-flexbox controls_list_theme-undefined
                
                undefined
              "
  >
    <div
      class="
                        controls-MasterDetail_master undefined
                        controls-MasterDetail_master-contrast controls-MasterDetail_master_width
                      "
      style="width: 200px; max-width: 299px; min-width: 250px;"
    >
      <div />
    </div>
    <div
      class="
                            controls-MasterDetail_details
                            controls-MasterDetail_details-contrast
                            ws-flexbox ws-flex-grow-1 
                          "
    >
      <div />
    </div>
    <div
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Controls/masterDetail:Base MasterDetail рендер с установкой ширины + propStorageId 1`] = `
<div>
  <div
    class="
                controls-MasterDetail controls-MasterDetail_masterPosition-left
                
                ws-flexbox controls_list_theme-undefined
                
                undefined
              "
  >
    <div
      class="
                        controls-MasterDetail_master undefined
                        controls-MasterDetail_master-contrast controls-MasterDetail_master_width
                      "
      style="width: 200px; max-width: 299px; min-width: 100px;"
    >
      <div />
    </div>
    <div
      class="
                        controls-MasterDetail_resizing-line_wrapper
                        controls-MasterDetail_resizingLine_masterPosition-left
                    "
    >
      <div
        class="controls-ResizingLine controls_dragnDrop_theme-default controls-MasterDetail_resizing-line
                            controls-MasterDetail_resizing-line_position-left"
        data-qa="controls-MasterDetail_resizing-line"
        tabindex="0"
      >
        <div
          class="controls-ResizingLine__wrapper"
        >
          <div
            class="controls-ResizingLine__areaWrapper controls-ResizingLine__areaWrapper_horizontal controls-ResizingLine__areaWrapper_horizontal_direction-direct"
          />
          <div
            class="controls-ResizingLine__border controls-ResizingLine__border_horizontal"
            tabindex="0"
          />
        </div>
      </div>
    </div>
    <div
      class="
                            controls-MasterDetail_details
                            controls-MasterDetail_details-contrast
                            ws-flexbox ws-flex-grow-1 
                          "
    >
      <div />
    </div>
    <div
      tabindex="0"
    />
  </div>
</div>
`;
